tests/tests.py

```
@@ -672,3 +673,89 @@ def test_wildcard_slash_replacement(self):
+        validate_slurm_extra(job)
+
+    def test_valid_slurm_extra(self, mock_job):
+        """Test that validation passes with allowed SLURM options."""
+        job = mock_job(slurm_extra="--mail-type=END --mail-user=user@example.com")
+        # Should not raise any exception
+        validate_slurm_extra(job)
+
+    def test_forbidden_job_name_long_form(self, mock_job):
+        """Test that --job-name is rejected."""
+        job = mock_job(slurm_extra="--job-name=my-job --mail-type=END")
+        with pytest.raises(WorkflowError, match="job-name.*not allowed"):
```

Ruff (0.13.1)

Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)